### PR TITLE
Add a ContainerUpdate consumer to listen for updates to a Container.

### PR DESCRIPTION
**Pull request by ankagrawal**
Add a listener to update a container's quota when a container is created or updated in AccountService.

### CODE REVIEW

**Comment by justinlin-linkedin** [comment]
We already have account updater consumers in AccountService, which exposes the same information and invoke the callback at the same time. i don't see a reason to add container updater consumers here.

**Comment by cgtz** [comment]
> We already have account updater consumers in AccountService, which exposes the same information and invoke the callback at the same time. i don't see a reason to add container updater consumers here.

Also, if there is ever an account-level quota feature in the future, we may want to react to account property changes in addition to container property changes.

File: ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java [hunk 1]
```diff
@@ -90,6 +90,7 @@
  *   back to zookeeper.
  * </p>
  */
+// TODO ANKUR why does it both extend AbstractAccountService and implement AccountService??
```
Comment (lightningrob) [inline ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java:93]: I probably just forgot to remove the implement clause.  Feel free to.

File: ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java [hunk 2]
```diff
@@ -326,6 +328,8 @@ public void updateAccounts(Collection<Account> accounts) throws AccountServiceEx
     }
     // Tell the world
     publishChangeNotice();
+    containerUpdateConsumers.forEach(containerUpdateConsumer -> containerUpdateConsumer.accept(
```
Comment (lightningrob) [inline ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java:331]: These calls should be made in HelixAccountService too.  Consider adding `AbstractAccountService.notifyAccountUpdateConsumers` and calling from both impls.

**Pull request closed without merging**